### PR TITLE
Add battery info

### PR DESCRIPTION
**Pull request by KazuCocoa**
- [x] add android
- [x] add doc
- [x] add changelog
- [x] test with the real server and add func tests

_Add battery information accessors #83_

### CODE REVIEW

File: lib/appium_lib_core/android/device.rb [hunk 1]
```diff
@@ -382,10 +382,26 @@ def get_performance_data(package_name:, data_type:, data_read_timeout: 1000)
...
           add_clipboard
           Emulator.emulator_commands
+          add_battery_info
         end
 
         private
 
+        def add_battery_info
+          ::Appium::Core::Device.add_endpoint_method(:get_battery_info) do
+            def get_clipboard(content_type: :plaintext)
```
Comment (mykola-mokhnach) [inline lib/appium_lib_core/android/device.rb]: copypaste?

Comment (KazuCocoa) [reply lib/appium_lib_core/android/device.rb]: oops, yup...
I forgot the change... (I've implemented the only iOS so far...)

File: test/unit/android/device_w3c_test.rb [hunk 2]
```diff
@@ -731,6 +731,18 @@ def test_image_comparison_get_images_similarity
 
         assert_requested(:post, "#{SESSION}/appium/compare_images", times: 1)
       end
+
+      def test_get_battery_info
+        stub_request(:post, "#{SESSION}/execute/sync")
+          .with(body: { script: 'mobile: batteryInfo', args: [{}] }.to_json)
+          .to_return(headers: HEADER, status: 200, body: { value: { state: 5, level: 0.5 } }.to_json)
```
Comment (KazuCocoa) [inline test/unit/android/device_w3c_test.rb]: If the `value` has `status`, then Ruby client raise parse response error.
I guess the implementation is https://github.com/SeleniumHQ/selenium/wiki/JsonWireProtocol#responses 's one.

```
  test_get_battery_info                                          ERROR (0.02s)
Selenium::WebDriver::Error::NoCollectionError:         Selenium::WebDriver::Error::NoCollectionError: unknown error, status=2: nil
            /Users/kazuaki/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.11.0/lib/selenium/webdriver/remote/response.rb:69:in `assert_ok'
            /Users/kazuaki/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.11.0/lib/selenium/webdriver/remote/response.rb:32:in `initialize'
            /Users/kazuaki/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.11.0/lib/selenium/webdriver/remote/http/common.rb:81:in `new'
            /Users/kazuaki/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.11.0/lib/selenium/webdriver/remote/http/common.rb:81:in `create_response'
            /Users/kazuaki/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.11.0/lib/selenium/webdriver/remote/http/default.rb:104:in `request'
            /Users/kazuaki/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.11.0/lib/selenium/webdriver/remote/http/common.rb:59:in `call'
            /Users/kazuaki/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.11.0/lib/selenium/webdriver/remote/bridge.rb:164:in `execute'
            /Users/kazuaki/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.11.0/lib/selenium/webdriver/remote/oss/bridge.rb:579:in `execute'
            /Users/kazuaki/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.11.0/lib/selenium/webdriver/remote/oss/bridge.rb:267:in `execute_script'
            /Users/kazuaki/GitHub/ruby_lib_core/lib/appium_lib_core/android/uiautomator2/device.rb:15:in `battery_info'
            /Users/kazuaki/GitHub/ruby_lib_core/test/unit/android/device_test.rb:763:in `test_get_battery_info'
```

**Comment by KazuCocoa** [comment]
image

Applied every related PRs in my locale and ensured they worked fine. For Android, I changed emulator settings and made sure them.

I'll merge after the CI 🍏

**Comment by KazuCocoa** [comment]
cc @mykola-mokhnach 
I've checked applying the Android related PRs in my locale and ✅ they work fine. ^

**Comment by mykola-mokhnach** [comment]
Thanks @KazuCocoa

**Pull request merged**